File: Offer/C++/offer21.cc
#include<iostream>
#include<vector>
using namespace std;

class Solution {
public:
    vector<int> exchange(vector<int>& nums) {
        int l = 0,  r = nums.size()-1;
        while(l < r){
            while(l < r && (nums[l] & 1) == 1) l++;
            while(l < r && (nums[r] & 1) == 0) r--;
            swap(nums[l],nums[r]);
        }
        return nums;
    }
};

File: Offer/src/Array/Offer21.java
package Array;

public class Offer21 {
    public int[] exchange(int[] nums) {
        int left = 0,right = nums.length-1;
        while (left < right){
            while(left < right && (nums[left] & 1) == 1) left++;
            while(left < right && (nums[right] & 1) == 0) right--;
            int  temp  = nums[left];
            nums[left] = nums[right];
            nums[right]  = temp;
        }
        return nums;
    }

    public void swap(int i,  int j){
        int  temp = i;
        i = j;
        j = temp;
    }
}
